cli/main.rs

```diff
@@ -178,114 +181,273 @@ fn print_cache_info(state: &GlobalState, json: bool) -> Result<(), ErrBox> {
-    println!("{} {}", colors::bold("type:"), output.file_type);
-    if let Some(compiled) = output.compiled {
+    println!("{} {}", colors::bold("local:"), local);
+    println!("{} {}", colors::bold("type:"), file_type);
+    if let Some(compiled) = compiled {
       println!("{} {}", colors::bold("compiled:"), compiled);
     }
-    if let Some(map) = output.map {
+    if let Some(map) = map {
       println!("{} {}", colors::bold("map:"), map);
     }
-    if let Some(deps) = output.deps {
-      println!("{}{}", colors::bold("deps:\n"), deps.name);
-      if let Some(ref depsdeps) = deps.deps {
-        for d in depsdeps {
-          println!("{}", d);
-        }
-      }
-    } else {
-      println!(
-        "{} cannot retrieve full dependency graph",
-        colors::bold("deps:"),
-      );
+
+    println!(
+      "{} {} unique",
+      colors::bold("deps:"),
+      &module_graph.len() - 1
+    );
+    println!(
+      "{} ({}, total = {})",
+      file_info.name,
+      human_size(file_info.size as f64),
+      human_size(file_info.total_size.unwrap() as f64)
+    );
+
+    for (idx, dep) in file_info.deps.iter().enumerate() {
+      print_file_dep_info(&dep, "  ", idx == file_info.deps.len() - 1);
     }
+
     Ok(())
   }
 }
 
+/// Prints the `FileInfoDepTree` tree to stdout.
+fn print_file_dep_info(info: &FileInfoDepTree, prefix: &str, is_last: bool) {
+  print_dep(prefix, is_last, info);
+
+  let prefix = &get_new_prefix(prefix, is_last);
+  let child_count = info.deps.len();
+  for (idx, dep) in info.deps.iter().enumerate() {
+    print_file_dep_info(dep, prefix, idx == child_count - 1);
+  }
+}
+
+/// Prints a single `FileInfoDepTree` to stdout.
+fn print_dep(prefix: &str, is_last: bool, info: &FileInfoDepTree) {
+  let has_children = !info.deps.is_empty();
+
+  println!(
+    "{}{}─{} {}{}",
+    prefix,
+    get_sibling_connector(is_last),
+    get_child_connector(has_children),
+    info.name,
+    get_formatted_totals(info)
+  );
+}
+
+/// Gets the formatted totals for the provided `FileInfoDepTree`.
+///
+/// If the total size is reported as 0 then an empty string is returned.
+fn get_formatted_totals(info: &FileInfoDepTree) -> String {
+  if let Some(total_size) = info.total_size {
+    format!(
+      " ({}, total = {})",
+      human_size(info.size as f64),
+      human_size(total_size as f64)
+    )
+  } else {
+    "".to_string()
+  }
+}
+
+/// Gets the sibling portion of the tree branch.
+fn get_sibling_connector(is_last: bool) -> char {
+  if is_last {
+    '└'
+  } else {
+    '├'
+  }
+}
+
+/// Gets the child connector for the branch.
+fn get_child_connector(has_children: bool) -> char {
+  if has_children {
+    '┬'
+  } else {
+    '─'
+  }
+}
+
+/// Creates a new prefix for a dependency tree item.
+fn get_new_prefix(prefix: &str, is_last: bool) -> String {
+  let mut prefix = prefix.to_string();
+  if is_last {
+    prefix.push(' ');
+  } else {
+    prefix.push('│');
+  }
+
+  prefix.push(' ');
+  prefix
+}
```

Comment:
Are those bits duplicated from `core/modules.rs::Deps`?

Reply:
Yes. I used Deps as a reference for the printing of the readable tree.